### ISSUE TITLE
Upgrade then Unable to create File System Image

### ISSUE THREAD

**Issue by djensen47** [docker/for-win#1514]
### Expected behavior
Docker for Windows starts after an update and doesn't say "EDGE" on the application icon.

### Actual behavior
Docker for WIndows doesn't start AT ALL. Even after an uninstall and re-install. Also, the app icon says "edge"

### Information

Windows 10 Pro Insider Preview 1709 Build 17063.1000
Docker: 17.12.0-ce-win46 (15048)

  - Diagnostic ID: 242D0BE4-373D-4E94-B9B1-6B21212875C9/2018-01-09_17-53-48

### Steps to reproduce the behavior

Upgrade to 

### Error Message
```
Unable to create File System Image
   at Docker.Core.Pipe.NamedPipeClient.Send(String action, Object[] parameters) in C:\gopath\src\github.com\docker\pinata\win\src\Docker.Core\pipe\NamedPipeClient.cs:line 36
   at Docker.Actions.DoStart(SynchronizationContext syncCtx, Boolean showWelcomeWindow, Boolean executeAfterStartCleanup) in C:\gopath\src\github.com\docker\pinata\win\src\Docker.Windows\Actions.cs:line 95
   at Docker.Actions.<>c__DisplayClass18_0.<Start>b__0() in C:\gopath\src\github.com\docker\pinata\win\src\Docker.Windows\Actions.cs:line 81
   at Docker.WPF.TaskQueue.<>c__DisplayClass19_0.<.ctor>b__1() in C:\gopath\src\github.com\docker\pinata\win\src\Docker.WPF\TaskQueue.cs:line 59
```

### Logs
```
Version: 17.12.0-ce-win46 (15048)
Channel: stable
Sha1: 0ac7325e4f480d6458b9d4ef12b50d8fcf782fae
Started on: 2018/01/09 17:46:35.833
Resources: C:\Program Files\Docker\Docker\Resources
OS: Windows 10 Enterprise Insider Preview
Edition: Professional
Id: 1709
Build: 17063
BuildLabName: 17063.1000.amd64fre.rs_prerelease.171213-1610
File: C:\Users\david\AppData\Local\Docker\log.txt
CommandLine: "C:\Program Files\Docker\Docker\Docker for Windows.exe" 
You can send feedback, including this log file, at https://github.com/docker/for-win/issues
[17:46:35.985][GUI            ][Info   ] Starting...
[17:46:36.705][Tracking       ][Info   ] Crash report and usage statistics are enabled
[17:46:36.709][SegmentApi     ][Info   ] Usage statistic: appLaunched
[17:46:38.430][NPS            ][Info   ] NPS user ceated on :10/4/2017 9:12:37 AM -07:00, last survey : 11/3/2017 11:35:01 AM -07:00
[17:46:38.431][SegmentApi     ][Info   ] Usage statistic: heartbeat
[17:46:38.434][SegmentApi     ][Info   ] Usage statistic: Identify
[17:46:38.570][NamedPipeClient][Info   ] Sending Version()...
[17:46:50.349][NamedPipeClient][Info   ] Sending Version()...
[17:46:50.416][NamedPipeClient][Info   ] Received response for Version
[17:46:50.419][NamedPipeClient][Info   ] Sending Start(Docker.Core.Settings, Docker.Core.DaemonOptions)...
[17:46:48.395][BackendServer  ][Info   ] Started
[17:46:50.413][NamedPipeServer][Info   ] Version()
[17:46:50.414][NamedPipeServer][Info   ] Version done in 00:00:00.0005018.
[17:46:50.428][NamedPipeServer][Info   ] Start(Docker.Core.Settings, Docker.Core.DaemonOptions)
[17:46:50.439][PowerMode      ][Info   ] Stop
[17:46:50.451][HyperV         ][Info   ] Stop
[17:46:50.464][PowerShell     ][Info   ] Run script with parameters: -Stop True...
[17:46:50.469][PowerShell     ][Info   ] Creating a Runspace Pool...
[17:46:50.584][PowerShell     ][Info   ] Runspace Pool created (Min=1, Max=2)
[17:46:50.668][HyperV         ][Info   ] Script started at 17:46:50.664
[17:46:52.883][HyperV         ][Info   ] Module loaded at 17:46:52.883
[17:46:53.144][HyperV         ][Info   ] VM MobyLinuxVM does not exist
[17:46:53.150][HyperV         ][Debug  ] [stop] took 00:00:02.6935298 to run
[17:46:53.151][OptimizeDisk   ][Info   ] Optimize
[17:46:53.152][PowerShell     ][Info   ] Run script...
[17:46:53.220][Moby           ][Info   ] Stop
[17:46:53.233][HyperV         ][Info   ] Create
[17:46:53.233][PowerShell     ][Info   ] Run script with parameters: -Create True -VhdPathOverride  -SwitchSubnetAddress 10.0.75.0 -SwitchSubnetMaskSize 24 -CPUs 2 -Memory 2048 -IsoFile C:\Program Files\Docker\Docker\Resources\docker-for-win.iso...
[17:46:53.235][PowerMode      ][Info   ] Start
[17:46:53.239][HyperVGuids    ][Info   ] Installing GUIDs...
[17:46:53.239][Firewall       ][Info   ] Removing all existing rules...
[17:46:53.242][HyperVGuids    ][Info   ] GUIDs installed
[17:46:53.259][HyperV         ][Info   ] Script started at 17:46:53.259
[17:46:53.282][HyperV         ][Info   ] Module loaded at 17:46:53.281
[17:46:53.336][Firewall       ][Info   ] All existing rules are removed.
[17:46:53.336][Firewall       ][Info   ] Opening ports for C:\Program Files\Docker\Docker\Resources\com.docker.proxy.exe...
[17:46:53.468][Firewall       ][Info   ] Opening ports for SMB...
[17:46:53.512][Firewall       ][Info   ] Ports are opened
[17:46:53.712][HyperV         ][Info   ] Creating Switch: DockerNAT...
[17:46:55.970][HyperV         ][Info   ] Switch created.
[17:46:58.603][HyperV         ][Info   ] Set IP address on switch
[17:46:59.756][HyperV         ][Info   ] Creating VM MobyLinuxVM...
[17:47:01.287][HyperV         ][Info   ] Setting CPUs to 2 and Memory to 2048 MB
[17:47:01.806][HyperV         ][Info   ] Attach VHD C:\Users\Public\Documents\Hyper-V\Virtual Hard Disks\MobyLinuxVM.vhdx
[17:47:02.712][HyperV         ][Info   ] Connect Internal Switch DockerNAT
[17:47:03.386][HyperV         ][Info   ] Attach DVD C:\Program Files\Docker\Docker\Resources\docker-for-win.iso
[17:47:04.724][HyperV         ][Info   ] Disabled Guest Service Interface
[17:47:04.782][HyperV         ][Info   ] Enabled Heartbeat
[17:47:04.855][HyperV         ][Info   ] Disabled Key-Value Pair Exchange
[17:47:04.893][HyperV         ][Info   ] Enabled Shutdown
[17:47:04.925][HyperV         ][Info   ] Enabled Time Synchronization
[17:47:04.972][HyperV         ][Info   ] Disabled VSS
[17:47:05.537][HyperV         ][Info   ] VM created.
[17:47:05.537][HyperV         ][Debug  ] [create] took 00:00:12.3041722 to run
[17:47:05.538][PowerShell     ][Info   ] Run script '(Hyper-V\Get-VM MobyLinuxVM).Id.Guid'...
[17:47:05.752][Database       ][Info   ] Generating CA Cert Bundle
[17:47:05.839][Database       ][Info   ] CA Cert Bundle Generated
[17:47:05.929][Linux          ][Error  ] Failed to start: Unable to create File System Image. Let's retry
[17:47:06.418][HyperV         ][Info   ] Hyper-V is running
[17:47:06.418][PowerMode      ][Info   ] Stop
[17:47:06.419][HyperV         ][Info   ] Stop
[17:47:06.419][PowerShell     ][Info   ] Run script with parameters: -Stop True...
[17:47:06.448][HyperV         ][Info   ] Script started at 17:47:06.448
[17:47:06.468][HyperV         ][Info   ] Module loaded at 17:47:06.468
[17:47:06.516][HyperV         ][Info   ] VM MobyLinuxVM is stopped
[17:47:06.516][HyperV         ][Debug  ] [stop] took 00:00:00.0969985 to run
[17:47:06.516][OptimizeDisk   ][Info   ] Optimize
[17:47:06.516][PowerShell     ][Info   ] Run script...
[17:47:08.176][Moby           ][Info   ] Stop
[17:47:08.212][HyperV         ][Info   ] Destroy
[17:47:08.216][PowerShell     ][Info   ] Run script with parameters: -Destroy True -KeepVolume True...
[17:47:08.230][HyperV         ][Info   ] Script started at 17:47:08.230
[17:47:08.284][HyperV         ][Info   ] Module loaded at 17:47:08.284
[17:47:08.391][HyperV         ][Info   ] VM MobyLinuxVM is stopped
[17:47:08.392][HyperV         ][Info   ] Destroying Switch DockerNAT...
[17:47:11.108][HyperV         ][Info   ] Removing VM MobyLinuxVM...
[17:47:11.354][HyperV         ][Debug  ] [destroy] took 00:00:03.1419019 to run
[17:47:11.354][Firewall       ][Info   ] Closing ports...
[17:47:11.354][Firewall       ][Info   ] Removing all existing rules...
[17:47:11.579][Firewall       ][Info   ] All existing rules are removed.
[17:47:11.579][Firewall       ][Info   ] Ports are closed
[17:47:11.580][HyperVGuids    ][Info   ] Removing GUIDs...
[17:47:11.584][HyperVGuids    ][Info   ] GUIDs removed
[17:47:11.584][HyperV         ][Info   ] Create
[17:47:11.584][PowerShell     ][Info   ] Run script with parameters: -Create True -VhdPathOverride  -SwitchSubnetAddress 10.0.75.0 -SwitchSubnetMaskSize 24 -CPUs 2 -Memory 2048 -IsoFile C:\Program Files\Docker\Docker\Resources\docker-for-win.iso...
[17:47:11.619][HyperV         ][Info   ] Script started at 17:47:11.619
[17:47:11.654][HyperV         ][Info   ] Module loaded at 17:47:11.654
[17:47:11.827][HyperV         ][Info   ] Creating Switch: DockerNAT...
[17:47:14.013][HyperV         ][Info   ] Switch created.
[17:47:16.201][HyperV         ][Info   ] Set IP address on switch
[17:47:16.221][HyperV         ][Info   ] Creating VM MobyLinuxVM...
[17:47:17.114][HyperV         ][Info   ] Setting CPUs to 2 and Memory to 2048 MB
[17:47:17.578][HyperV         ][Info   ] Attach VHD C:\Users\Public\Documents\Hyper-V\Virtual Hard Disks\MobyLinuxVM.vhdx
[17:47:18.227][HyperV         ][Info   ] Connect Internal Switch DockerNAT
[17:47:18.513][HyperV         ][Info   ] Attach DVD C:\Program Files\Docker\Docker\Resources\docker-for-win.iso
[17:47:19.065][HyperV         ][Info   ] Disabled Guest Service Interface
[17:47:19.086][HyperV         ][Info   ] Enabled Heartbeat
[17:47:19.116][HyperV         ][Info   ] Disabled Key-Value Pair Exchange
[17:47:19.136][HyperV         ][Info   ] Enabled Shutdown
[17:47:19.167][HyperV         ][Info   ] Enabled Time Synchronization
[17:47:19.200][HyperV         ][Info   ] Disabled VSS
[17:47:19.850][HyperV         ][Info   ] VM created.
[17:47:19.850][HyperV         ][Debug  ] [create] took 00:00:08.2663199 to run
[17:47:19.850][PowerMode      ][Info   ] Stop
[17:47:19.850][HyperV         ][Info   ] Stop
[17:47:19.850][PowerShell     ][Info   ] Run script with parameters: -Stop True...
[17:47:19.872][HyperV         ][Info   ] Script started at 17:47:19.872
[17:47:19.887][HyperV         ][Info   ] Module loaded at 17:47:19.887
[17:47:19.945][HyperV         ][Info   ] VM MobyLinuxVM is stopped
[17:47:19.945][HyperV         ][Debug  ] [stop] took 00:00:00.0950071 to run
[17:47:19.945][OptimizeDisk   ][Info   ] Optimize
[17:47:19.945][PowerShell     ][Info   ] Run script...
[17:47:21.333][Moby           ][Info   ] Stop
[17:47:21.356][HyperV         ][Info   ] Create
[17:47:21.356][PowerShell     ][Info   ] Run script with parameters: -Create True -VhdPathOverride  -SwitchSubnetAddress 10.0.75.0 -SwitchSubnetMaskSize 24 -CPUs 2 -Memory 2048 -IsoFile C:\Program Files\Docker\Docker\Resources\docker-for-win.iso...
[17:47:21.357][HyperVGuids    ][Info   ] Installing GUIDs...
[17:47:21.357][PowerMode      ][Info   ] Start
[17:47:21.357][Firewall       ][Info   ] Removing all existing rules...
[17:47:21.358][HyperVGuids    ][Info   ] GUIDs installed
[17:47:21.391][HyperV         ][Info   ] Script started at 17:47:21.390
[17:47:21.424][HyperV         ][Info   ] Module loaded at 17:47:21.424
[17:47:21.440][Firewall       ][Info   ] All existing rules are removed.
[17:47:21.440][Firewall       ][Info   ] Opening ports for C:\Program Files\Docker\Docker\Resources\com.docker.proxy.exe...
[17:47:21.450][Firewall       ][Info   ] Opening ports for SMB...
[17:47:21.458][Firewall       ][Info   ] Ports are opened
[17:47:21.774][HyperV         ][Info   ] Using existing Switch: DockerNAT
[17:47:26.452][HyperV         ][Info   ] Using existing Switch IP address
[17:47:26.500][HyperV         ][Info   ] Setting CPUs to 2 and Memory to 2048 MB
[17:47:27.060][HyperV         ][Info   ] Connect Internal Switch DockerNAT
[17:47:27.480][HyperV         ][Info   ] Remove existing DVDs
[17:47:27.956][HyperV         ][Info   ] Attach DVD C:\Program Files\Docker\Docker\Resources\docker-for-win.iso
[17:47:28.613][HyperV         ][Info   ] Disabled Guest Service Interface
[17:47:28.641][HyperV         ][Info   ] Enabled Heartbeat
[17:47:28.705][HyperV         ][Info   ] Disabled Key-Value Pair Exchange
[17:47:28.756][HyperV         ][Info   ] Enabled Shutdown
[17:47:28.788][HyperV         ][Info   ] Enabled Time Synchronization
[17:47:28.828][HyperV         ][Info   ] Disabled VSS
[17:47:29.020][HyperV         ][Info   ] VM created.
[17:47:29.020][HyperV         ][Debug  ] [create] took 00:00:07.6642901 to run
[17:47:29.020][PowerShell     ][Info   ] Run script '(Hyper-V\Get-VM MobyLinuxVM).Id.Guid'...
[17:47:29.045][Database       ][Info   ] Generating CA Cert Bundle
[17:47:29.103][Database       ][Info   ] CA Cert Bundle Generated
[17:47:29.107][PowerMode      ][Info   ] Stop
[17:47:29.107][HyperV         ][Info   ] Stop
[17:47:29.107][PowerShell     ][Info   ] Run script with parameters: -Stop True...
[17:47:29.125][HyperV         ][Info   ] Script started at 17:47:29.125
[17:47:29.139][HyperV         ][Info   ] Module loaded at 17:47:29.138
[17:47:29.165][HyperV         ][Info   ] VM MobyLinuxVM is stopped
[17:47:29.165][HyperV         ][Debug  ] [stop] took 00:00:00.0580003 to run
[17:47:29.165][OptimizeDisk   ][Info   ] Optimize
[17:47:29.165][PowerShell     ][Info   ] Run script...
[17:47:30.299][Moby           ][Info   ] Stop
[17:47:30.306][HyperV         ][Info   ] Destroy
[17:47:30.306][PowerShell     ][Info   ] Run script with parameters: -Destroy True -KeepVolume True...
[17:47:30.324][HyperV         ][Info   ] Script started at 17:47:30.324
[17:47:30.334][HyperV         ][Info   ] Module loaded at 17:47:30.334
[17:47:30.355][HyperV         ][Info   ] VM MobyLinuxVM is stopped
[17:47:30.356][HyperV         ][Info   ] Destroying Switch DockerNAT...
[17:47:32.269][HyperV         ][Info   ] Removing VM MobyLinuxVM...
[17:47:32.359][HyperV         ][Debug  ] [destroy] took 00:00:02.0530037 to run
[17:47:32.359][Firewall       ][Info   ] Closing ports...
[17:47:32.359][Firewall       ][Info   ] Removing all existing rules...
[17:47:32.474][NamedPipeClient][Error  ] Unable to send Start: Unable to create File System Image
[17:47:32.477][Notifications  ][Error  ] Unable to create File System Image
[17:47:32.463][Firewall       ][Info   ] All existing rules are removed.
[17:47:32.463][Firewall       ][Info   ] Ports are closed
[17:47:32.463][HyperVGuids    ][Info   ] Removing GUIDs...
[17:47:32.464][HyperVGuids    ][Info   ] GUIDs removed
[17:47:32.471][NamedPipeServer][Error  ] Unable to execute Start: Unable to create File System Image    at Docker.Backend.ContainerEngine.Linux.DoStart(Settings settings, String daemonOptions) in C:\gopath\src\github.com\docker\pinata\win\src\Docker.Backend\ContainerEngine\Linux.cs:line 256
   at Docker.Backend.ContainerEngine.Linux.Start(Settings settings, String daemonOptions) in C:\gopath\src\github.com\docker\pinata\win\src\Docker.Backend\ContainerEngine\Linux.cs:line 130
   at Docker.Core.Pipe.NamedPipeServer.<>c__DisplayClass9_0.<Register>b__0(Object[] parameters) in C:\gopath\src\github.com\docker\pinata\win\src\Docker.Core\pipe\NamedPipeServer.cs:line 47
   at Docker.Core.Pipe.NamedPipeServer.RunAction(String action, Object[] parameters) in C:\gopath\src\github.com\docker\pinata\win\src\Docker.Core\pipe\NamedPipeServer.cs:line 145
[17:48:30.148][ErrorReportWindow][Info   ] Open logs

```

**Comment by djensen47** [comment]
Possible duplicate of #1221 and #1458

**Comment by justinmchase** [comment]
I have the same issue. I uninstalled and reinstalled and still get the same issue. This came immediately after the latest update and renders docker completely unusable on windows.

**Comment by taureliloome** [comment]
Hi, had the same problem, seems one of the suggested solutions on possible duplicates solved the problem for me:
  Enable-WindowsOptionalFeature -Online -FeatureName Microsoft-Hyper-V -All
Run that in elevated power shell

**Comment by Jagoop** [comment]
I am experiencing the same bug on both my laptop and my desktop PC. The main similarities is same build of Windows: 17074

**Comment by bronxct1** [comment]
Also seeing this same issue on build 17074. Diagnostic ID 30165A10-397B-49A0-9A53-1ACD1BB261D9/2018-01-17_12-01-35.

I also cannot manually enable shared drives in the settings.

**Comment by selner** [comment]
+1 on the upgrade blocking Docker from starting at all.  

- Diagnostic report submitted:  ID#8F946141-A26C-4E2C-AAB2-CA2D338AE49B/2018-01-23_16-49-16
- Version: 17.12.0-ce-win47 (15139)
- Windows 10: Build 17074

**Comment by RobertRelyea** [comment]
Same issue on 17074, reverted to 17025 and docker is fine.

**Comment by taylorb-microsoft** [comment]
We have narrowed this down to something related to the IFileSystemImage.  We are still looking at what change could have caused it to break but we are getting closer.

**Comment by PatrickLang** [comment]
I'm tracking this internally - MS#15594197 . Here's another example that breaks:
https://gist.github.com/PatrickLang/2166c70e997791f643b9bb2ee8350b8d

**Comment by laytoneverson** [comment]
If this helps another diagnostic: 24DDD2B5-0880-4E0C-8264-6A9B1AA18EC0/2018-01-24_01-41-35

**Comment by AceHack** [comment]
I have this error an in the windows event logs I see
WARN: Unable to locate plugin: internal
ERROR: Error occurred when creating network plugin not found

**Comment by ahaeber** [comment]
I managed to start the MobyLinuxVM using the MobyLinux.ps1 powershell script. There was an error at line 301 because it should not use Test-Path if $confIsoFile is null / empty string. After I added a -ne "" check it works. Like this:
`    if ($confIsoFile -ne "") {
        if (Test-Path $confIsoFile) {
            if ((Get-Item $confIsoFile).length -gt 0) {
                Write-Output "Attach Config ISO $confIsoFile"
                if (($vm | Get-VMScsiController).length -le 1) {
                    $vm | Add-VMScsiController
                }
                $vm | Hyper-V\Add-VMDvdDrive -ControllerNumber 1 -ControllerLocation 1 -Path $confIsoFile
            }
        }
    }
`
(Probably the -ne "" check can be combined with Test-Path to avoid nesting if's)

Although the VM starts in Hyper-V the rest of Docker does not start. So it does not fix all isues.

**Comment by kamkie** [comment]
Same for me on windows 17083 C2316E66-0B1E-48AF-BC4E-36B7BE02E4EA/2018-01-24_22-22-52

**Comment by calielc** [comment]
I had the same problem with Windows 10 (Build 17083) and docker versions 17.12 and 18.01

I did the downgrade to docker 17.09.1 and workds!! https://download.docker.com/win/stable/14687/Docker%20for%20Windows%20Installer.exe

**Comment by taylorb-microsoft** [comment]
We think the underlying Windows bug has been identified and we have a fix in review.  Once the PR has been merged, verified and a target build number has been identified I will update this issue.  I have requested it be fast tracked so we can flight out the fix ASAP.

**Comment by NoobTW** [comment]
Windows 10 Build 17074. Uninstall and re-install 17.09.1 works, too.

**Comment by hach-que** [comment]
Latest edge release is still broken in 17083. Workaround is to use Docker Toolbox.

**Comment by dannyk81** [comment]
It will be broken until the underlying Windows [bug](https://github.com/docker/for-win/issues/1514#issuecomment-360889478) is resolved.

There were two bugs in Windows, the first (https://github.com/moby/moby/issues/35854) was fixed in 17083, this one (Unable to create File System Image) is still open.

@taylorb-microsoft any updates on this? 😄 really want to play with the new built-in Kubernetes feature in Edge version.

**Comment by dogopupper** [comment]
since 17.09.1 works, how to download this older version?

**Comment by pd93** [comment]
@dogopupper You can find download links for older versions on the [release notes page](https://docs.docker.com/docker-for-windows/release-notes/#docker-community-edition-17091-ce-win42-2017-12-11-stable). However, it appears there is no link for any version over `17.09`. I've commented on [this issue](https://github.com/docker/for-win/issues/162) asking why.

**Comment by dannyk81** [comment]
Windows Build 17093 was [released](https://blogs.windows.com/windowsexperience/2018/02/07/announcing-windows-10-insider-preview-build-17093-pc/#BwMLhmMiyZmX1m4B.97), however [reports](https://github.com/docker/for-win/issues/1458#issuecomment-364077052) are saying that same problem persists 😢 

The release note indeed does not mention anything related to Docker...

@taylorb-microsoft would you be able to tell if this build was supposed to fix this issue?

**Comment by PatrickLang** [comment]
@dannyk81 sorry, but the fix was incomplete. You may see this error instead now, but it's still the same broken component in Windows.

Docker UI will show:
```
Binary stream '0' does not contain a valid BinaryHeader. Possible causes are invalid stream or object version change between serialization and deserialization.
   at Docker.Core.Pipe.NamedPipeClient.Send(String action, Object[] parameters) in C:\gopath\src\github.com\docker\pinata\win\src\Docker.Core\pipe\NamedPipeClient.cs:line 36
   at Docker.Actions.DoStart(SynchronizationContext syncCtx, Boolean showWelcomeWindow, Boolean executeAfterStartCleanup) in C:\gopath\src\github.com\docker\pinata\win\src\Docker.Windows\Actions.cs:line 95
   at Docker.Actions.<>c__DisplayClass18_0.<Start>b__0() in C:\gopath\src\github.com\docker\pinata\win\src\Docker.Windows\Actions.cs:line 81
   at Docker.WPF.TaskQueue.<>c__DisplayClass19_0.<.ctor>b__1() in C:\gopath\src\github.com\docker\pinata\win\src\Docker.WPF\TaskQueue.cs:line 59
```


You'll find a crash report in the Windows application event log:
```
Log Name:      Application
Source:        Application Error
Date:          2/7/2018 10:05:48 PM
Event ID:      1000
Task Category: (100)
Level:         Error
Keywords:      Classic
User:          N/A
Computer:      vagrant-10
Description:
Faulting application name: com.docker.service, version: 17.12.0.15139, time stamp: 0x5a580556
Faulting module name: ntdll.dll, version: 10.0.17093.1000, time stamp: 0x1fc8f560
```

**Comment by dannyk81** [comment]
Thanks @PatrickLang!

**Comment by nevi-me** [comment]
Is the only solution to revert to an older Docker version?

**Comment by jlost** [comment]
Windows 10 Pro Build 180116
```
λ docker --version
Docker version 17.12.0-ce, build c97c6d6
```
Diagnostic: 073A60F1-EDDF-4F16-88FF-B04DC99ED137/2018-02-09_07-25-57

**Comment by arjenpdevries** [comment]
Same problem on Build 17093 prerelease 180202-1400.

**Comment by SonicGD** [comment]
Still doesn't work on new 17604 build

**Comment by dannyk81** [comment]
@PatrickLang @taylorb-microsoft looks like 17101 & 17604 also don't have the fix, can you confirm?

**Comment by carlfischer1** [comment]
https://github.com/docker/for-win/issues/1540#issuecomment-366051772 for update on workaround in D4W

**Comment by PatrickLang** [comment]
17604 is "skip ahead" and doesn't have the fix yet. Still waiting on build number for that branch.
Builds >= 17104 and < 17600 should be ok with any version of D4W

**Comment by dannyk81** [comment]
@PatrickLang I'm on 17101, how would I upgrade to 17104? what setting should I have for this?

**Comment by haroldvandekamp** [comment]
I am on Windows 10 Insider Slow version 1709 Build 17074.1002 and also had the error "Unable to create file system image" when I was using Docker Community Edition 17.12.0-ce-win47 2018-01-12 (Stable).
I uninstalled that version, downloaded and installed the Docker Community Edition 17.09.1-ce-win42 2017-12-11 (Stable) version (https://download.docker.com/win/stable/14687/Docker%20for%20Windows%20Installer.exe), after that Docker starts normally.

**Comment by jeskoriesner** [comment]
@haroldvandekamp worked for me as well. It just needed a restart for me to function properly. Thx.

**Comment by carloshenriqueribeiro** [comment]
### **+ 1**
**OS:** W10 Pro Insider 1709 (Build 17074.1002)
**Server:**
 Engine:
Version: 18.02.0-ce
API version:  1.36 (minimum version 1.24)
Go version: go1.9.3
Git commit: fc4de44
Built:  Wed Feb  7 21:26:24 2018
OS/Arch: windows/amd64
Experimental: true

**diagnostic id: 7B3C5A35-F5BA-4D55-AB33-DA588E416950/2018-02-23_19-57-16**

**Comment by pd93** [comment]
This is fixed in the latest fast ring build [17107] with Docker 18.02 (Edge)
See [this](https://github.com/docker/for-win/issues/1458#issuecomment-368133379) comment

**Comment by gmanolache** [comment]
+1 latest stable build is broken latest windows.

**Comment by rn** [comment]
We have changed the mechanism by which the filesystem is created and that fix is available in both stable and edge channels. Closing

**Comment by solox174** [comment]
I had this same issue on:
Version: 18.03.1-ce-win65 (17513)
Channel: stable
Sha1: 93354b38bd7210badafff8e0ce1f0be9a465be83
Started on: 2018/06/28 19:31:22.996
Resources: C:\Program Files\Docker\Docker\Resources
OS: Windows 10 Enterprise
Edition: Enterprise
Id: 1703
Build: 15063
BuildLabName: 15063.0.amd64fre.rs2_release.170317-1834

Deleting all of the Hyper-v related network adapters using the device manager and letting Docker re-create them fixed it.
